src/libmw/include/mw/mmr/LeafIndex.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+// Copyright (c) 2018-2019 David Burkett
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include <mw/common/Macros.h>
+#include <mw/mmr/Index.h>
+#include <mw/util/BitUtil.h>
+
+MMR_NAMESPACE
+
+class LeafIndex
+{
+public:
+    LeafIndex() noexcept
+        : m_leafIndex(0), m_nodeIndex(0, 0) { }
+    LeafIndex(const uint64_t leafIndex, const uint64_t position)
+        : m_leafIndex(leafIndex), m_nodeIndex(position, 0) { }
```

Comment:
Shouldn't it use Index::At(position) here?

Reply:
No, if the height is already known, we should use the constructor directly to avoid the unnecessary (and moderately expensive) height calculation.